ToolChain/Assembler/Parser/Parser.py

```diff
@@ -20,636 +19,383 @@
-                    buildInstruction, lineInstruction = self._parseCodeLine(line, rawLine)
-
-                    if buildInstruction is not None:
-                        # There was a valid instruction or memory reference on the line
-
-                        if type(buildInstruction.instructionCode) is str:
-                            # We have a memory location reference
-                            memoryReferenceDict[buildInstruction.instructionCode] = relativeAddressCounter
-                        else:
-                            # We have an instruction
-                            instructionList.append(lineInstruction)
-                            relativeAddressCounter += buildInstruction.instructionLength
-
-            except Exception as e:
-                # If all else fails, the line has invalid code
-                print("ERROR while asssembling line {}: '{}'".format(str(lineNo), line,))
-                print(e)
-                quit()
-
-        return instructionList, memoryReferenceDict, relativeAddressCounter
-
-    def _parseCodeLine(self, line: list=None, rawLine: str=None):
-        """
-        This will parse an extracted line of code and build an instruction from that code line
-        Parsing a line of code requires many steps.
-
-            1- Find the possible codes for the mnemonic from OperationDescription
-            2- Based on the possible codes and parts of the lines, find the correct form
-            3- Based on the form, build the instruction
-            4- Return the instruction
-                If the instruction returned has a string as instruction mnemonic, this means that the
-                instruction simply represent a label in the code. It will be removed from the final
-                program.
-
-        :param line: line split into individual arguments
-        :param rawLine: raw line as we originally read it from the file, used to preserve spaces in .dataAlpha fields
-        :return: The instruction object with all info about arguments, and the line's binary instruction
-        """
+        line = text.split()                     # We split the line with space as a delimiter. Each token is an argument
+        instruction = b""                       # Instruction will be written as a bytestring
+        arglist = []                            # We keep each argument in a list to evaluate later
+        found = False                           # Flag to indicate if instruction was found in state list
+        form = "Ins"                            # We assume the first argument is an instruction
+        labelFlag = 0                           # Used by the assembler to determine what we're returning
+        dataIdentifier = line[0].upper()        # used for special identifiers such as .dataAlpha, comments etc.
+        self.lineno += 1
+
+        if (dataIdentifier in IDENTIFIER_LIST) or (dataIdentifier[0] in IDENTIFIER_LIST) or \
```

Comment:
If splitting line like this, I would do 3 lines with 1 condition on each instead of 1 line with 2 and then 1 line with a single one.